### ISSUE TITLE
Ghostown - Sushi Boyz : Scrolling logo issue

### ISSUE THREAD

**Issue by emoon** [dirkwhoffmann/vAmiga#390]
In this demo https://demozoo.org/productions/136336/ https://files.scene.org/get/parties/2015/revision15/demo_amiga/gtn-sushiboyz-party.adf

At 1:58 there should be text coming in from the right to the left (see this youtube ref) http://www.youtube.com/watch?v=msRTT5f8BGg&t=1m58s then there should be a logo scrolling in the middle but in vAmiga it gets stuck on the left side. Also the credits names that comes in looks incorrect as well

![000](https://user-images.githubusercontent.com/350822/88460285-5e907680-ce9b-11ea-8ec3-622c98c6b791.png)

I have validated that it works correct on original A500

**Comment by dirkwhoffmann** [comment]
Hmm, yes, the version on YouTube looks better 🤔.

Here is an X-ray and the relevant part of the Copper list: 

<img width="927" alt="Bildschirmfoto 2020-07-26 um 15 24 10" src="https://user-images.githubusercontent.com/12561945/88480237-1b92d980-cf55-11ea-89bf-e6271add3aa6.png">

<img width="385" alt="Bildschirmfoto 2020-07-26 um 15 29 58" src="https://user-images.githubusercontent.com/12561945/88480248-26e60500-cf55-11ea-9524-4cffc458db63.png">

To me, this looks like the BPLxPT registers start incrementing too early. All I can say for sure at the moment is that there are no sprites involved.

**Comment by dirkwhoffmann** [comment]
Interesting. The scene is broken in UAE, too, but it works well on the real machine.

<img width="681" alt="Bildschirmfoto 2020-08-01 um 15 29 36" src="https://user-images.githubusercontent.com/12561945/89102702-f4805000-d40b-11ea-9552-0015bd1eb058.png">
<img width="1128" alt="Bildschirmfoto 2020-08-01 um 15 30 12" src="https://user-images.githubusercontent.com/12561945/89102705-f8ac6d80-d40b-11ea-828f-77f58f29e81f.png">

I'll add @tonioni to this thread.

UAE and vAmiga seem to make the exact same error here...

**Comment by emoon** [comment]
Ah that is interesting. I only tested on my original A500 for comparision.

**Comment by emoon** [comment]
I think the youtube capture is done in WinUAE tho. Maybe it requires some specific setting?

**Comment by tonioni** [comment]
It is OCS/ECS difference. There probably is also fixed version available so make sure everyone uses same version.

**Comment by dirkwhoffmann** [comment]
> It is OCS/ECS difference

Confirmed. It runs perfectly well with an ECS Agnus in UAE. In vAmiga, I did select an ECS Agnus, so there must be some ECS feature that is still emulated the OCS way in my emulator.

**Comment by emoon** [comment]
but my A500 is OCS and it works there?

**Comment by dirkwhoffmann** [comment]
> but my A500 is OCS and it works there?

Seems like a quest for star commander @mithrendal with his old spaceship, the A1000.

**Comment by mithrendal** [comment]
So you guys want me to unearth the old rusty spaceship again. ✌️ It will be a pleasure 🧑🏻‍🚀

It was part of the federal terranian space war fleet long ago ... and it was its flagship 👩🏿‍🚀

Is the demo working with 512 KB only ?

**Comment by emoon** [comment]
I think it needs 1 MB of ram

**Comment by dirkwhoffmann** [comment]
> I think it needs 1 MB of ram

Yes, doesn't run with 512KB, only 🙁.

<img width="877" alt="Bildschirmfoto 2020-08-01 um 16 22 32" src="https://user-images.githubusercontent.com/12561945/89103611-45477700-d413-11ea-8767-fd8b39d3cf64.png">

**Comment by mithrendal** [comment]
Ohh 😭 ... then we leave A1000 spaceship in the nagar

**Comment by dirkwhoffmann** [comment]
I've recorded some debugging output for line 236 (the line where the parameters for the bottom area are set):

```
...
[13394] (236,122) C0E14C  0 -CBS-A 6020 07C1 Agnus: setBPLxPTL(4, 19F8)
[13394] (236,132) C0E14C  0 -CBS-A 6020 07C1 [0614C8] Agnus: pokeBPLCON1(88)
[13394] (236,132) C0E14C  0 -CBS-A 6020 07C1 [0614C8] Denise: pokeBPLCON1(88)
[13394] (236,133) C0E14C  0 -CBS-A 6020 07C1 Agnus: setBPLCON1(0,88)
[13394] (236,133) C0E14C  0 -CBS-A 6020 07C1 Denise: setBPLCON1(88)
[13394] (236,140) C0E14C  0 -CBS-A 6020 07C1 [0614D0] Agnus: pokeBPL1MOD(2E)
[13394] (236,142) C0E14C  0 -CBS-A 6020 07C1 Agnus: setBPL1MOD(2E)
[13394] (236,144) C0E14C  0 -CBS-A 6020 07C1 [0614D4] Agnus: pokeBPL2MOD(2E)
[13394] (236,146) C0E14C  0 -CBS-A 6020 07C1 Agnus: setBPL2MOD(2E)
[13394] (236,148) C0E14C  0 -CBS-A 6020 07C1 [0614D8] Agnus: pokeBPLCON1(0)
[13394] (236,148) C0E14C  0 -CBS-A 6020 07C1 [0614D8] Denise: pokeBPLCON1(0)
[13394] (236,149) C0E14C  0 -CBS-A 6020 07C1 Agnus: setBPLCON1(88,0)
[13394] (236,149) C0E14C  0 -CBS-A 6020 07C1 Denise: setBPLCON1(0)
[13394] (236,160) C0E14C  0 -CBS-A 6020 07C1 [0614E0] Agnus: pokeDMACON(8100)
[13394] (236,162) C0E14C  0 -CBS-A 6020 07C1 Agnus: setDMACON(6ef, 8100)
[13394] (236,162) C0E14C  0 BCBS-A 6020 07C1 Denise: Bitplane DMA switched on
```

I guess the issue has something to do with bitplane DMA being switched on in the middle of the line (hpos = 162).

**Comment by tonioni** [comment]
You probably don't have the original party version if it does not glitch on OCS A500. (I just retested and the scroller does glitch on 8371 Agnus A500)

And yes, bitplane DMA mid scanline works differently on OCS vs ECS/AGA.

I didn't check but it probably is same side-effect as in demo Subtle Shades / Nuance. (Credits part)

**Comment by emoon** [comment]
Ok, I'm going to double check :)

**Comment by emoon** [comment]
So https://files.scene.org/get/parties/2015/revision15/demo_amiga/gtn-sushiboyz-party.adf linked above works fine on my A500 🤔

**Comment by tonioni** [comment]
ADF I tested had CRC32=b2b78929. And it did have the problem, scroller is misaligned towards left. You really sure you do have 8371 Agnus? :)

**Comment by emoon** [comment]
CRC matches. Are there more than one revision of the OCS chip? I'm 100% sure it's not an ECS machine.

**Comment by tonioni** [comment]
Perhaps there is simple misunderstanding: I am talking about ECS Agnus only, not "full" ECS (which usually means A500+ or A600 or A3000 unless someone swapped Denise) hardware that includes both ECS Denise and Agnus. Agnus being OCS or ECS is the important bit. (But some demos also do have issues with ECS Denise: Screen is totally blank..)

Run SysInfo (or any other system info program). Does it say "DMA/Gfx ECS Agnus" or not?

**Comment by emoon** [comment]
It actually say ECS Agnus indeed :)

**Comment by tonioni** [comment]
Time to buy another A500, OCS and ECS A500 is the minimum requirement :D

**Comment by mithrendal** [comment]
@tonioni do you think 512kb is not enough anymore  in 2020? please don’t tell me that I have to buy a ram expansion 😬

**Comment by dirkwhoffmann** [comment]
> do you think 512kb is not enough anymore in 2020?

ACA500plus 🤤. Works like a charm.

**Comment by dirkwhoffmann** [comment]
Fixed in v0.9.10

<img width="1018" alt="Bildschirmfoto 2020-08-03 um 15 50 52" src="https://user-images.githubusercontent.com/12561945/89190513-df3d2a00-d5a1-11ea-9d4b-6f4a3b4940d4.png">
